### ISSUE TITLE
Roll YCgCo-R out

### ISSUE THREAD

**Issue by vrabaud** [AOMediaCodec/libavif#2077]
Since it has been approved (but not published yet as of 25/03/2024, cf https://www.itu.int/rec/T-REC-H.273), we can rool YCgCo-R.
We first need to roll the decoder out by default, and once there are enough clients, we can roll the encoder out by default.

**Comment by vrabaud** [comment]
Waiting on https://www.itu.int/rec/T-REC-H.273 to be public. Looking at the draft at https://jvet-experts.org/doc_end_user/documents/30_Antalya/wg11/JVET-AD1008-v1.zip (JVET-AF1008), it seems the enums have changed to 16 and 17.

Note that equations E-14 and E-15 have one error and they need to be subtractions instead of additions (BitdepthY - 2 and BitdepthY - 1).

**Comment by wantehchang** [comment]
[H.273](https://www.itu.int/rec/T-REC-H.273) (09/23) has been published. However, it is already in the suspended state, because it will be replaced by H.273 (07/24).

In H.273 (09/23) the values of YCgCo-Re and YCgCo-Ro are 16 and 17, respectively. So we will need to change `AVIF_MATRIX_COEFFICIENTS_YCGCO_RE` and `AVIF_MATRIX_COEFFICIENTS_YCGCO_RO` to the new values.

**Comment by wantehchang** [comment]
After consulting with other members of the AOMedia Image Container subgroup, we will go ahead and promote YCgCo-Re and YCgCo-Ro to libavif's stable API based on [H.273](https://www.itu.int/rec/T-REC-H.273) (09/23).

I suggest we plan to make this change early next week, after the Chrome 129 release branch is created. However, if this is low risk, we don't need to wait.

@vigneshvg Vignesh: do we need to change CrabbyAvif at the same time?

**Comment by vigneshvg** [comment]
> @vigneshvg Vignesh: do we need to change CrabbyAvif at the same time?

yes, there probably has to be some minor changes. i will take a look at it and make the changes.

**Comment by y-guyon** [comment]
Freely available H.273 V4 (07/24) mentions:
| Value | Matrix coefficients |
|-|-|
|16 |YCgCo-Re|
|17| YCgCo-Ro|

See https://www.itu.int/rec/T-REC-H.273-202407-I/en.

**Comment by novomesk** [comment]
Hello,
I am encountering an interesting issue in Google Chrome Version 143.0.7499.170 (Official Build) (64-bit)

I tried to make a YCgCo-Re AVIF image:

http://188.121.162.14/girl.avif

When I open the link for a first time, the colors look OK:

<img width="913" height="987" alt="Image" src="https://github.com/user-attachments/assets/a6d5561e-56a6-4a24-ab58-882767afa4ef" />

However, when I press F5 in the browser for several times, it happens I see different color version:

<img width="913" height="987" alt="Image" src="https://github.com/user-attachments/assets/21f566f2-38c8-497f-8798-fee9a9e7c485" />

**Comment by vrabaud** [comment]
I am able to reproduce your bug. That is really weird. This is probably on the Chrome side with caching. Or maybe a race condition in libavif ?

**Comment by FooIbar** [comment]
Didn't Chrome switch to CrabbyAvif?
This does seem like a bug on the Chrome side, it always gives wrong colors if I right-click the image and select "open image in new tab"
And I found that even when the image loads with the correct colors, the thumbnails in the DevTools' "Sources" and "Network" tabs still have the wrong colors.

**Comment by vrabaud** [comment]
@vigneshvg , any opinion ?

**Comment by vigneshvg** [comment]
> @vigneshvg , any opinion ?

i will take a look.

**Comment by vigneshvg** [comment]
i was able to reproduce the issue and as i thought the difference is because of the two different decoding paths that are possible in chrome's image pipeline. ycgco-r is only supported via crabbyavif and not when decoding to yuv directly. i've uploaded a fix to chromium here: https://chromium-review.googlesource.com/c/chromium/src/+/7399312
